app/controllers/snippets_controller.rb

```diff
@@ -77,8 +77,9 @@ def set_vault
     end
 
     def set_snippet
-      @vault = Vault.find(params[:vault_id])
-      @snippet = @vault.snippets.find(params[:id])
+      @vault = Vault.find_by_id(params[:vault_id])
```

Comment:
Extracting `VaultsController#set_vault` into the `ApplicationController` might be a good idea, as it allows that method to be shared to the `SnippetsController`.